src/cbc_sdk/platform/alerts.py

```diff
@@ -997,6 +998,132 @@ def set_rows(self, rows):
         self._batch_size = rows
         return self
 
+    def set_time_range(self, *args, **kwargs):
+        """
+        For v7 Alerts:
+
+        Sets the 'time_range' query body parameter, determining a time range based on 'backend_timestamp'.
+
+        Args:
+            *args: not used
+            **kwargs (dict): Used to specify start= for start time, end= for end time, and range= for range. Values are
+            either timestamp ISO 8601 strings or datetime objects for start and end time. For range the time range to
+            execute the result search, ending on the current time. Should be in the form "-2w",
+            where y=year, w=week, d=day, h=hour, m=minute, s=second.
+
+        For v6 Alerts (backwards compatibility):
+
+        Restricts the alerts that this query is performed on to the specified time range for a given key. Will also set
+        the 'time_range' as in the v7 usage if key is create_time or backend_timestamp. Will be deprecated with v6 alert
+        api
+
+        Args:
+            key (str): The key to use for criteria one of create_time, first_event_time, last_event_time,
+             backend_timestamp, backend_update_timestamp, or last_update_time
+            **kwargs (dict): Used to specify start= for start time, end= for end time, and range= for range. Values are
+            either timestamp ISO 8601 strings or datetime objects for start and end time. For range the time range to
+            execute the result search, ending on the current time. Should be in the form "-2w",
+            where y=year, w=week, d=day, h=hour, m=minute, s=second.
+
+
+        Returns:
+            AlertSearchQuery: This instance.
+
+        Examples:
+            >>> query = api.select(Alert).set_time_range(start="2020-10-20T20:34:07Z")
+            >>> second_query = api.select(Alert).
+            ...     set_time_range(start="2020-10-20T20:34:07Z", end="2020-10-30T20:34:07Z")
+            >>> third_query = api.select(Alert).set_time_range(range='-3d')
+
+        """
+        args_count = args.__len__()
+        time_filter = self._create_valid_time_filter(kwargs)
+        if args_count > 0:
+            key = args[0]
+            if self._is_valid_time_criteria_key(key):
+                self.add_time_criteria(key, **kwargs)
```

Comment:
key needs to be converted from v6 field name to v7.
Would like to see a mock test for this that validates the post request.

This python snippet:
alert_list = cb.select(Alert).set_time_range("first_event_time", start="2023-09-19T01:00:00.000", end="2023-09-20T01:00:00.000")

should generate this request.  (rows, start are set in base.  Get latest off the parent branch to remove the empty query element):
{
    "criteria": {
        "first_event_timestamp": {
            "end": "2023-09-20T01:00:00.000Z",
            "start": "2023-09-19T01:00:00.000Z"
        }
    },
    "rows": 100,
    "start": 1
}